src/main/xar-resources/data/urlrewrite/urlrewrite.xml

```diff
@@ -10,175 +8,169 @@
-          </para>
-          <para>For instance:
-            <literal>example.xml</literal>.</para>
+          <para>The <code>XQueryUrlRewrite</code> servlet looks for an XQuery script to interpret the rest of the URI (see <xref linkend="controller-mappings"/>).  By convention this is called <code>controller.xq</code> (or in older applications <code>controller.xql</code>).  The default configuration will work with a script saved with this name in the base collection of an application.</para>
         </listitem>
-      </varlistentry>
-      <varlistentry>
-        <term>
-          <code>exist:controller</code>
-        </term>
         <listitem>
-          <para>The part of the URI leading to the current controller script.</para>
-          <para>For example: if the request path is
-            <literal>/xquery/test.xql</literal>
-            and the controller is in the
-            <literal>xquery</literal>
-            collection,
-            <literal>$exist:controller</literal>
-            will contain
-            <literal>/xquery</literal>.</para>
+          <para><code>controller.xq</code> examines the URI using pre-defined <xref linkend="variables"/>, and produces a fragment in the <xref linkend="controller-xml"/>.</para>
         </listitem>
-      </varlistentry>
-      <varlistentry>
-        <term>
-          <code>exist:prefix</code>
-        </term>
         <listitem>
-          <para>If the current controller hierarchy is mapped to a certain path prefix,
-            <literal>$exist:prefix</literal>
-            returns that prefix.</para>
-          <para>
-            For example: the default configuration maps the path
-            <literal>/tools</literal>
-            to a collection in the database (see below). In this case,
-            <literal>$exist:prefix</literal>
-            would contain
-            <literal>/tools</literal>.</para>
+          <para><code>XQueryURLRewrite</code> uses the controller XML fragment as an instruction on what to do next.  This instruction may be as simple as forwarding to a resource on (or off) the server, or it may be as complex as a pipeline using the <emphasis>Model-View-Controller</emphasis> pattern (see <xref linkend="mvc-pipelines"/>) and other servlets such as eXist's <xref linkend="xq-servlet"/> or <xref linkend="xslt-servlet"/>.</para>
         </listitem>
-      </varlistentry>
-      <varlistentry>
-        <term>
-          <code>exist:root</code>
-        </term>
+      </orderedlist>
+    <sect2 xml:id="eg1">
+      <title>Example I: A Simple Implementation</title>
+      <para>Consider  the document you are currently reading; a direct URL pointing to the source data might be <code>/exist/apps/doc/data/urlrewrite.xml</code>.  But accessing this URI  would only show a user the raw XML: instead users should get a properly formatted HTML version of the text, accessible through a simple URL like <code>/exist/apps/doc/urlrewrite</code>.</para>
```

Comment:
I think the direct source URL to the source XML for this article would be `/exist/rest/db/apps/doc/data/urlrewrite.xml`.

Reply:
I will reword this, since the URLs do not really point to this exact document (which is at `/exist/apps/doc/data/urlrewrite/urlrewrite.xml` - I think bringing the rest interface into it muddies the waters even further).  I considered changing all of the URLs to match the actual docbook article, but I think the important thing is that they are consistent.